internal/destination/destination.go

```diff
@@ -168,18 +170,34 @@ func (d *Destination) Open(ctx context.Context) error {
 }
 
 // Write writes a record into a Destination.
-func (d *Destination) Write(_ context.Context, records []sdk.Record) (int, error) {
-	if !d.nc.IsConnected() {
-		return 0, errWriteUnavailable
-	}
-
-	for i, record := range records {
-		if err := d.writer.Write(record); err != nil {
-			return i, fmt.Errorf("write: %w", err)
+func (d *Destination) Write(ctx context.Context, records []sdk.Record) (int, error) {
+	timeout, cancel := context.WithTimeout(ctx, d.config.RetryWait*time.Duration(d.config.RetryAttempts))
```

Comment:
I am changing the connector to set defaults if values are set to zero. Reasoning [with opinion]: If it was just pure NATS, not Jetstream, fire-and-forget is a desirable feature, but it is not the case for Jetstream.

We have a situation of a middle-man (connector) who can't rely on the host (conduit:  fires the write command and puts it in a waiting group, but doesn't timeout) or the NATS server where we don't have how to know which version and that could result in dangling connections.

Reply:
That's fine. I'm still wondering why we need a timeout context here? The context is going to make the number of retries non-deterministic, or in the worst case even cause the connector to fail the write. Imagine that we want to retry once (`RetryAttempts=1`) and wait 1 second between retries (`RetryWait=1s`), so the timeout will be set to 1 second. If the connection is slow or we have a big batch it might take more than 1 second to send all records, so we could hit the timeout before we manage to send everything.